terraform/modules/single_instance/data.tf

```
@@ -116,7 +116,8 @@ data "openstack_networking_network_v2" "public" {
   name = "public"
 }
 data "openstack_networking_floatingip_v2" "public" {
-  pool = data.openstack_networking_network_v2.public.id
+  address = var.floatingip_address
+  #pool = data.openstack_networking_network_v2.public.id
```

To make this optional, we can write it like this:
```suggestion
  pool = var.floatingip_address == null ? data.openstack_networking_network_v2.public.id : null
  address = var.floatingip_address # Null by default
```
That way it will either select the only address in the pool (or throw an error) _or_ it will use the address from the variable.